src/test/java/ir/sahab/nexus/plugin/tag/internal/dto/IntegrationTest.java

```diff
@@ -0,0 +1,222 @@
+            .build();
+
+    private Client client;
+    private WebTarget target;
+
+    private AssociatedComponent component1;
+    private AssociatedComponent component2;
+
+    @Before
+    public void setup() {
+        client = ClientBuilder.newClient();
+        target = client.target("http://nexus:8081/service/rest" + APIConstants.V1_API_PREFIX);
+        component1 = new AssociatedComponent(REPO_MAVEN_RELEASES, "gr1", "comp1", randomAlphanumeric(5));
+        uploadMavenComponent(component1);
+        component2 = new AssociatedComponent(REPO_MAVEN_RELEASES, "gr2", "comp2", randomAlphanumeric(5));
+        uploadMavenComponent(component2);
+        //TODO: Test non-maven artifacts which does not have group/version
+    }
+
+
+    @After
+    public void tearDown() {
+        client.close();
+    }
+
+    private void uploadMavenComponent(AssociatedComponent component) {
+        MultipartFormDataOutput output = new MultipartFormDataOutput();
+        output.addFormData("maven2.groupId", component.getGroup(), MediaType.TEXT_PLAIN_TYPE);
+        output.addFormData("maven2.artifactId", component.getName(), MediaType.TEXT_PLAIN_TYPE);
+        output.addFormData("maven2.version", component.getVersion(), MediaType.TEXT_PLAIN_TYPE);
+        output.addFormData("maven2.asset1.extension", "jar", MediaType.TEXT_PLAIN_TYPE);
+        ByteArrayInputStream inputStream = new ByteArrayInputStream(randomAlphanumeric(100).getBytes());
+        output.addFormData("maven2.asset1", inputStream, MediaType.APPLICATION_OCTET_STREAM_TYPE,
+                "test-maven-artifact.jar");
+
+        Response response = target.path("components").queryParam("repository", component.getRepository())
+                .request()
+                .header(HttpHeaders.AUTHORIZATION, authorizationHeader())
+                .post(Entity.entity(output, MediaType.MULTIPART_FORM_DATA_TYPE));
+        assertEquals(Family.SUCCESSFUL, response.getStatusInfo().getFamily());
+    }
+
+    /**
+     * @return authorization header of basic authentication
+     */
+    private static String authorizationHeader() {
+        String auth = USERNAME + ":" + PASSWORD;
+        byte[] encodedAuth = Base64.encodeBase64(auth.getBytes(StandardCharsets.ISO_8859_1));
+        return "Basic " + new String(encodedAuth);
+    }
+
+    @Test
+    public void testCrud() {
+        Map<String, String> attributes = new HashMap<>();
+        attributes.put(CHANGE_ID, randomAlphanumeric(20));
+        attributes.put(STATUS, "failed");
+        attributes.put("Commit-Id", randomAlphanumeric(20));
+
+        TagDefinition tag = new TagDefinition(randomAlphanumeric(5), attributes, Arrays.asList(component1));
+
+        // Add Tag
+        Response response = addTag(tag);
+        assertEquals(Family.SUCCESSFUL, response.getStatusInfo().getFamily());
+        Tag postResponseTag = response.readEntity(Tag.class);
+        assertFalse(new Date().before(postResponseTag.getFirstCreated()));
```

Comment:
use java 8 time functions

Reply:
Why we should do this? I haven't used new java API in order to have json serialization and dependency issues, so why should I use those APIs to compare old Date instances and add complexity to it?